examples/watchasay/test_watchasay.py

```diff
@@ -24,7 +24,7 @@
 from globus_action_provider_tools.testing.fixtures import flask_helpers_noauth
 
 
-@pytest.fixture(scope="module")
+@pytest.fixture
 def client(flask_helpers_noauth):
```

Comment:
This change was due entirely to the fact that it pulls in the `flask_helpers_noauth` fixture, which was session-scoped. The moment that thing gets used, it patches a method of a class, which prevents all of the `globus_sdk._testing` mocking from ever getting touched.

No problem! I can make `flask_helpers_noauth` function-scoped so the class method isn't perma-mocked for the entire duration of the test suite. Argh, but pytest complains that change is incompatible because this `client` fixture remains module-scoped. Pytest fixtures cannot be more broadly-scoped than the fixtures they depend on.

Tightening the scope of `flask_helpers_noauth` forced everything that used it to also tighten up.

I didn't measure overhead or timing differences. However, the per-Python test suite takes ~1.5 seconds to execute.

Reply:
I'm going to mark this as resolved but please re-open if this needs more review.

Reply:
Yeah this isn't something I know.

>However, the per-Python test suite takes ~1.5 seconds to execute.

This nullifies all my concerns though. If we're running a full test suite in 1.5 seconds, we don't need to be concerned about optimizing it further.